lib/LoginStrategy.ts

```diff
@@ -239,4 +253,49 @@ export class LoginStrategy extends BaseStrategy {
       return acc
     }, [] as Claim[])
   }
-}
+
+  /**
+   * @description checks if the claimer is authenticated into the DID Document
+   *
+   * @param token
+   * @param payload
+   *
+   * @returns {string} issuer DID or null
+   */
+  async verifyDidClaim(token: string, payload: ITokenPayload): Promise<string> {
+
+    const didDocument = await this.didResolver.read(payload.iss)
+
+    if (this.isAuthorized(token, didDocument))
+      return payload.iss
+    return null 
+  }
+
+  private isAuthorized = (claimedToken: string,didDocument: IDIDDocument) : boolean => {
+
+    //retrieve claim infos from token
+    const claimInfos = jwt.decode(claimedToken) as {[key: string]: any;}
+    const didPubKey = didDocument.publicKey
+
+    if (didPubKey.length === 0)
+      return false
+    
+    //extract all authenticated users and keep those refering issuer's publicKey
+    const publicKeys = didDocument.authentication.map((auth, index) => {
+      const auth_pubkeyRef = auth["publicKey"] //public key reference in authentication field
+      const pubkey_id = didPubKey[index]["id"] //public key Id in publickKey field of DID document
```

Comment:
I don't think `index` can be used as it is currently being used. I think it would fail for a DID document like this one, right?
```
  "authentication": [
    {
      "publicKey": "key-2"
    }
  ],
  "publicKey": [
    {
      "id": "key-1",
      "publicKeyHex": "<some public key>"
    },
    {
      "id": "key-2",
      "publicKeyHex": "<some public key>"
    }
  ]
```

Reply:
Yes, you are right. If key is in the `publicKey` array but not in the `authentication` array, then my understanding is that it is not a valid `authentication` key. 

That said, there is another scenario that I haven't brought up yet that we should accommodate 😅 
Basically, I think you've covered this scenario, where an authentication claim signed by a key like this one should pass:
```
{
  "id": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E",
  "service": [],
  "authentication": [
    {
      "type": "owner",
      "validity": {
        "_hex": "0x1fffffffffffff"
      },
      "publicKey": "my-favorite-public-key-1"
    }
  ],
  "publicKey": [
    {
      "id": "my-favorite-public-key-1",
      "type": "Secp256k1veriKey",
      "controller": "0x96142FAC151454E38B897cE50fdE26C60622B14E",
      "publicKeyHex": "0x035f58d27f27944f42089083a6c96b099360bb6099988bcfe3ddca74541c0a097d"
    }
  ]
}
```

There is also a second scenario:
```
{
  "id": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E",
  "service": [],
  "authentication": [],
  "publicKey": [
    {
      "id": "my-favorite-public-key-1",
      "type": "Secp256k1sigAuth",
      "controller": "0x96142FAC151454E38B897cE50fdE26C60622B14E",
      "publicKeyHex": "0x035f58d27f27944f42089083a6c96b099360bb6099988bcfe3ddca74541c0a097d"
    }
  ]
}
```
Basically, this key isn't listed in the `authentication` array but because it is the `sigAuth` type, it is valid authentication key.

(@JGiter I am correct in saying these two scenarios are what should constitute valid "authentication" keys?)

Reply:
@JGiter I have been wondering about this because our current DID document "creation" gives the key a `veriKey` type but then the key is added to the `authentication` array, which is contrary to what I described above... 🤔

Reply:
Very interesting, thank you John for pointing those different scenarios. Indeed, I already cover the scenario where a claimed is signed with that kind of key:
`"publicKey": [
    {
      "id": "my-favorite-public-key-1",
      "type": "Secp256k1veriKey",
      "controller": "0x96142FAC151454E38B897cE50fdE26C60622B14E",
      "publicKeyHex": "0x035f58d27f27944f42089083a6c96b099360bb6099988bcfe3ddca74541c0a097d"
    }
  ]`
  
  So I targuet on managing the scenario with a sigAuth type of key

Reply:
Found the mapping in the current DIF did-ethr-resolver that handles this case: https://github.com/decentralized-identity/ethr-did-resolver/blob/18f4488b2bffe8cd3939224020b386cecfea9c9f/src/resolver.ts#L215

If `sigAuth` is used, then they are adding to the `auth` array. So I think we are on the correct path.